draft-davidben-tls-merkle-tree-certs.md

```diff
@@ -1423,6 +1629,165 @@ at-trustAnchorID ATTRIBUTE ::= {
+
+A subtree consistency proof for `[start, end)` and the tree of `n` elements is very similar to an inclusion proof for element `end - 1`. Starting at this element, incorporating the whole inclusion proof should reconstruct `root_hash` and incorporating a subset of the inclusion proof should reconstruct `node_hash`. However, instead of starting at level 0, the proof can be optimized to start at a higher level, specifically the highest ancestor of `end - 1` that is also directly contained in the tree.
+
+{{fig-truncate-consistency-proof}} depicts a subtree consistency proof between the subtree `[0, 6)` and the Merkle Tree of size 7. The consistency proof begins at level 1, or node `[4, 6)`. Note that, although element 6 at level 0 is in the consistency proof, there is a corresponding skipped node at level 1, the starting level. To demonstrate this, the figure denotes skipped nodes with duplicate values.
+
+~~~aasvg
+       +----------------+
+       |     [0, 6)     |       level 3
+       +----------------+
+        /           |
+   +========+  +--------+
+   | [0, 4) |  | [4, 6) |       level 2
+   +========+  +--------+
+    /      \        |
++-----+ +-----+ +~~~~~+
+|[0,2)| |[2,4)| |[4,6)|         level 1
++-----+ +-----+ +~~~~~+
+  / \     / \     / \
++-+ +-+ +-+ +-+ +-+ +-+
+|0| |1| |2| |3| |4| |5|         level 0
++-+ +-+ +-+ +-+ +-+ +-+
+
+
+       +----------------+
+       |     [0, 7)     |       level 3
+       +----------------+
+        /              \
+   +========+      +--------+
+   | [0, 4) |      | [4, 7) |   level 2
+   +========+      +--------+
+    /      \        /    |
++-----+ +-----+ +~~~~~+ +=+
+|[0,2)| |[2,4)| |[4,6)| |6|     level 1
++-----+ +-----+ +~~~~~+ +=+
+  / \     / \     / \    |
++-+ +-+ +-+ +-+ +-+ +-+ +-+
+|0| |1| |2| |3| |4| |5| |6|     level 0
++-+ +-+ +-+ +-+ +-+ +-+ +-+
+~~~
+{: #fig-truncate-consistency-proof title="A subtree consistency proof that starts at level 1 instead of level 0"}
+
+The procedure is structured similarly to inclusion proof evaluation ({{inclusion-proof-bits}}). Step 2 initializes `fn` (first number), `sn` (second number), and `tn` (third number) to follow, respectively, the paths to `start`, `end - 1` (the last element of the subtree), and `n - 1` (the last element of the tree).
+
+Steps 3 and 4 then skip to the starting node, described above. The starting node may be:
+
+* The entire subtree `[start, end)` if `[start, end)` is directly contained in the tree. This will occur if `end` is `n`, or if `[start, end)` is full.
+
+* Otherwise, the highest full subtree along the right edge of `[start, end)`.
+
+In the first case, `fn` will equal `sn` after truncation. Step 5 will then initialize the hashes to `node_hash`. The consistency proof does not need to include a separate copy of `node_hash`.
+
+In the second case, `fn` is less than `sn`. Step 6 will then initialize the hashes to the first value in the consistency proof.
```

Comment:
It's not clear what cases these are referring to, and they're not the two bullet points above. (`fn` can equal `sn` after performing either of those steps.) I'd clarify by changing this to something like the following:

> Steps 5 and 6 initialize the hashes `fr` and `sr`.
>
> If `fn` equals `sn` (possibly after truncation), step 5 initializes the hashes to `node_hash` because the subtree `[start, end)` is directly contained in the tree, so the consistency proof did not need to include a separate copy of `node_hash`.
>
> Otherwise, `fn` is less than `sn`, and step 6 initializes the hashes to the first value in the consistency proof.